diff --git a/src/lib/plugin/desktopfileparser.cpp b/src/lib/plugin/desktopfileparser.cpp
index 1122af8..207099d 100644
--- a/src/lib/plugin/desktopfileparser.cpp
+++ b/src/lib/plugin/desktopfileparser.cpp
@@ -32,6 +32,7 @@
 #include <QJsonObject>
 #include <QMutex>
 #include <QStandardPaths>
+#include <QCoreApplication>
 
 #if QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
 // in the desktoptojson binary enable debug messages by default, in the library only warning messages
@@ -245,8 +246,18 @@ QVector<CustomPropertyDefinition>* parseServiceTypesFile(QString path)
         const QString originalPath = path;
         path = QStandardPaths::locate(QStandardPaths::GenericDataLocation,
                                QStringLiteral("kservicetypes5/") + path);
+
+        if (path.isEmpty()) {
+            QFileInfo info(QStringLiteral("%1/../share/kservicetypes5/%2").arg(QCoreApplication::applicationDirPath(), originalPath));
+            if (info.exists()) {
+                path = info.absoluteFilePath();
+            }
+        }
         if (path.isEmpty()) {
-            qCCritical(DESKTOPPARSER) << "Could not locate service type file kservicetypes5/" << originalPath << ", tried" << QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation);
+            qCCritical(DESKTOPPARSER) << "Could not locate service type file kservicetypes5/" << originalPath
+            << ", tried in:" << QStandardPaths::standardLocations(QStandardPaths::GenericDataLocation)
+            << "\nand ../share/kservicetypes5/"  << originalPath << " in:"
+            << QCoreApplication::applicationDirPath();
             return nullptr;
         }
     }
