diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0dc4064..21db006 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ ecm_setup_version(${KF5_VERSION} VARIABLE_PREFIX KCONFIGWIDGETS
                         PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF5ConfigWidgetsConfigVersion.cmake"
                         SOVERSION 5)
 
-find_package(KF5Auth ${KF5_DEP_VERSION} REQUIRED)
+#find_package(KF5Auth ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5CoreAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Codecs ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Config ${KF5_DEP_VERSION} REQUIRED)
@@ -42,7 +42,7 @@ remove_definitions(-DQT_NO_CAST_FROM_BYTEARRAY)
 add_definitions(-DTRANSLATION_DOMAIN=\"kconfigwidgets5\")
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
     ki18n_install(po)
-    kdoctools_install(po)
+    #kdoctools_install(po)
 
     file(GLOB lang_dirs "po/*")
     foreach(lang_dir ${lang_dirs})
diff --git a/KF5ConfigWidgetsConfig.cmake.in b/KF5ConfigWidgetsConfig.cmake.in
index 9096da6..e49ec05 100644
--- a/KF5ConfigWidgetsConfig.cmake.in
+++ b/KF5ConfigWidgetsConfig.cmake.in
@@ -1,6 +1,6 @@
 @PACKAGE_INIT@
 
-find_dependency(KF5Auth "@KF5_DEP_VERSION@")
+#find_dependency(KF5Auth "@KF5_DEP_VERSION@")
 find_dependency(KF5Codecs "@KF5_DEP_VERSION@")
 find_dependency(KF5Config "@KF5_DEP_VERSION@")
 find_dependency(KF5WidgetsAddons "@KF5_DEP_VERSION@")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 10862c6..23835d0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -5,7 +5,7 @@ set(kconfigwidgets_SRCS
   kconfigdialog.cpp
   kconfigviewstatesaver.cpp
   kconfigdialogmanager.cpp
-  kcmodule.cpp
+  #kcmodule.cpp
   khelpclient.cpp
   klanguagebutton.cpp
   kpastetextaction.cpp
@@ -25,7 +25,7 @@ target_link_libraries(KF5ConfigWidgets
     KF5::Codecs # KCodecActions uses KCharsets, KEncodingProber
     KF5::WidgetsAddons # For K*Action, KPage*, KViewStateSerializer, KAcceleratorManager, K*GuiItem
     KF5::ConfigGui # KStandardAction uses KStandardShortcut
-    KF5::Auth # KCModule uses KAuth::Action
+    #KF5::Auth # KCModule uses KAuth::Action
   PRIVATE
     Qt5::DBus # KPasteTextAction talks to Klipper via DBus
     KF5::CoreAddons # KCModule uses KAboutData, KTipDialog uses KRandom
@@ -46,7 +46,7 @@ ecm_generate_headers(KConfigWidgets_HEADERS
   KConfigDialog
   KConfigViewStateSaver
   KConfigDialogManager
-  KCModule
+  #KCModule
   KHelpClient
   KLanguageButton
   KPasteTextAction
