diff --git a/CMakeLists.txt b/CMakeLists.txt
index f61b210..2d98d66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,7 @@ find_package(KF5DBusAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5I18n ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Service ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5DocTools ${KF5_DEP_VERSION})
-find_package(KF5Solid ${KF5_DEP_VERSION} REQUIRED) # for kio_trash
+#find_package(KF5Solid ${KF5_DEP_VERSION} REQUIRED) # for kio_trash
 
 if (NOT KIOCORE_ONLY)
 find_package(KF5Bookmarks ${KF5_DEP_VERSION} REQUIRED)
@@ -103,7 +103,7 @@ add_subdirectory(src)
 add_subdirectory(autotests)
 
 if (NOT KIOCORE_ONLY)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 endif()
 
 # create a Config.cmake and a ConfigVersion.cmake file and install them
diff --git a/KF5KIOConfig.cmake.in b/KF5KIOConfig.cmake.in
index 0577c9a..eebdf98 100644
--- a/KF5KIOConfig.cmake.in
+++ b/KF5KIOConfig.cmake.in
@@ -14,7 +14,7 @@ find_dependency(KF5Bookmarks "@KF5_DEP_VERSION@")
 find_dependency(KF5Completion "@KF5_DEP_VERSION@")
 find_dependency(KF5ItemViews "@KF5_DEP_VERSION@")
 find_dependency(KF5JobWidgets "@KF5_DEP_VERSION@")
-find_dependency(KF5Solid "@KF5_DEP_VERSION@")
+#find_dependency(KF5Solid "@KF5_DEP_VERSION@")
 find_dependency(KF5XmlGui "@KF5_DEP_VERSION@")
 endif()
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f65ad8e..20f5ac1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -8,12 +8,12 @@ add_subdirectory(kssld)
 add_subdirectory(kioslave)
 
 if (NOT KIOCORE_ONLY)
-add_subdirectory(kpasswdserver)
+#add_subdirectory(kpasswdserver)
 add_subdirectory(widgets)
-add_subdirectory(filewidgets)
+#add_subdirectory(filewidgets)
 add_subdirectory(kioexec)
 add_subdirectory(urifilters)
-add_subdirectory(kcms)
+#add_subdirectory(kcms)
 add_subdirectory(new_file_templates)
 
 if(NOT WIN32)
diff --git a/src/ioslaves/CMakeLists.txt b/src/ioslaves/CMakeLists.txt
index 36f1ab0..c4ad8fb 100644
--- a/src/ioslaves/CMakeLists.txt
+++ b/src/ioslaves/CMakeLists.txt
@@ -1,10 +1,10 @@
 
 add_subdirectory( file )
-add_subdirectory( help )
+#add_subdirectory( help )
 add_subdirectory( http )
 add_subdirectory( ftp )
 add_subdirectory( mailto )
 add_subdirectory( telnet )
-add_subdirectory( trash )
+#add_subdirectory( trash )
 add_subdirectory( protocols )
 
diff --git a/src/urifilters/CMakeLists.txt b/src/urifilters/CMakeLists.txt
index a3d7eb1..c061fc5 100644
--- a/src/urifilters/CMakeLists.txt
+++ b/src/urifilters/CMakeLists.txt
@@ -1,4 +1,4 @@
 add_subdirectory(localdomain)
-add_subdirectory(ikws)
+#add_subdirectory(ikws)
 add_subdirectory(shorturi)
 add_subdirectory(fixhost)
