diff -urw kio-5.12.0/CMakeLists.txt KIO/CMakeLists.txt
--- kio-5.12.0/CMakeLists.txt	2015-07-05 12:06:31.000000000 +0300
+++ KIO/CMakeLists.txt	2015-07-31 19:51:25.254638899 +0300
@@ -40,8 +40,8 @@
 find_package(KF5DBusAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5I18n ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Service ${KF5_DEP_VERSION} REQUIRED)
-find_package(KF5DocTools ${KF5_DEP_VERSION} REQUIRED)
-find_package(KF5Solid ${KF5_DEP_VERSION} REQUIRED) # for kio_trash
+#find_package(KF5DocTools ${KF5_DEP_VERSION} REQUIRED)
+#find_package(KF5Solid ${KF5_DEP_VERSION} REQUIRED) # for kio_trash
 
 if (NOT KIOCORE_ONLY)
 find_package(KF5Bookmarks ${KF5_DEP_VERSION} REQUIRED)
@@ -82,7 +82,7 @@
 add_definitions(-DTRANSLATION_DOMAIN=\"kio5\")
 if (IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/po")
     ki18n_install(po)
-    kdoctools_install(po)
+    #kdoctools_install(po)
 endif()
 if (KF5DocTools_FOUND)
 add_subdirectory(docs)
@@ -92,7 +92,7 @@
 add_subdirectory(autotests)
 
 if (NOT KIOCORE_ONLY)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 endif()
 
 # create a Config.cmake and a ConfigVersion.cmake file and install them
diff -urw kio-5.12.0/KF5KIOConfig.cmake.in KIO/KF5KIOConfig.cmake.in
--- kio-5.12.0/KF5KIOConfig.cmake.in	2015-07-05 12:06:31.000000000 +0300
+++ KIO/KF5KIOConfig.cmake.in	2015-07-31 19:57:30.806453868 +0300
@@ -14,7 +14,7 @@
 find_dependency(KF5Completion "@KF5_DEP_VERSION@")
 find_dependency(KF5ItemViews "@KF5_DEP_VERSION@")
 find_dependency(KF5JobWidgets "@KF5_DEP_VERSION@")
-find_dependency(KF5Solid "@KF5_DEP_VERSION@")
+#find_dependency(KF5Solid "@KF5_DEP_VERSION@")
 find_dependency(KF5XmlGui "@KF5_DEP_VERSION@")
 endif()
 
diff -urw kio-5.12.0/src/CMakeLists.txt KIO/src/CMakeLists.txt
--- kio-5.12.0/src/CMakeLists.txt	2015-07-05 12:06:31.000000000 +0300
+++ KIO/src/CMakeLists.txt	2015-07-31 19:47:17.442650596 +0300
@@ -8,12 +8,12 @@
 add_subdirectory(kioslave)
 
 if (NOT KIOCORE_ONLY)
-add_subdirectory(kpasswdserver)
+#add_subdirectory(kpasswdserver)
 add_subdirectory(widgets)
-add_subdirectory(filewidgets)
+#add_subdirectory(filewidgets)
 add_subdirectory(kioexec)
 add_subdirectory(urifilters)
-add_subdirectory(kcms)
+#add_subdirectory(kcms)
 
 if(NOT WIN32)
   # arpa/nameser.h
diff -urw kio-5.12.0/src/core/kfileitem.cpp KIO/src/core/kfileitem.cpp
--- kio-5.12.0/src/core/kfileitem.cpp	2015-07-05 12:06:31.000000000 +0300
+++ KIO/src/core/kfileitem.cpp	2015-07-30 22:09:44.318788768 +0300
@@ -30,6 +30,7 @@
 #include <QMimeDatabase>
 #include <QDebug>
 #include <qplatformdefs.h>
+#include <QDataStream>
 
 #include <klocalizedstring.h>
 #include <kdesktopfile.h>
diff -urw kio-5.12.0/src/ioslaves/CMakeLists.txt KIO/src/ioslaves/CMakeLists.txt
--- kio-5.12.0/src/ioslaves/CMakeLists.txt	2015-07-05 12:06:31.000000000 +0300
+++ KIO/src/ioslaves/CMakeLists.txt	2015-07-30 22:09:44.350789035 +0300
@@ -1,10 +1,10 @@
 
 add_subdirectory( file )
-add_subdirectory( help )
+#add_subdirectory( help )
 add_subdirectory( http )
 add_subdirectory( ftp )
 add_subdirectory( mailto )
 add_subdirectory( telnet )
-add_subdirectory( trash )
+#add_subdirectory( trash )
 add_subdirectory( protocols )
 
diff -urw kio-5.12.0/src/urifilters/CMakeLists.txt KIO/src/urifilters/CMakeLists.txt
--- kio-5.12.0/src/urifilters/CMakeLists.txt	2015-07-05 12:06:31.000000000 +0300
+++ KIO/src/urifilters/CMakeLists.txt	2015-07-31 19:45:03.942656898 +0300
@@ -1,4 +1,4 @@
 add_subdirectory(localdomain)
-add_subdirectory(ikws)
+#add_subdirectory(ikws)
 add_subdirectory(shorturi)
 add_subdirectory(fixhost)
