diff -urw kxmlgui-5.12.0/CMakeLists.txt XmlGui/CMakeLists.txt
--- kxmlgui-5.12.0/CMakeLists.txt	2015-07-05 12:16:23.000000000 +0300
+++ XmlGui/CMakeLists.txt	2015-07-30 20:52:18.375730063 +0300
@@ -33,6 +33,7 @@
 set(REQUIRED_QT_VERSION 5.2.0)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets DBus Xml Network PrintSupport)
 
+find_package(KF5CoreAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5ItemViews ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5Config ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5GlobalAccel ${KF5_DEP_VERSION} REQUIRED)
diff -urw kxmlgui-5.12.0/src/CMakeLists.txt XmlGui/src/CMakeLists.txt
--- kxmlgui-5.12.0/src/CMakeLists.txt	2015-07-05 12:16:23.000000000 +0300
+++ XmlGui/src/CMakeLists.txt	2015-07-30 20:53:51.568984248 +0300
@@ -78,6 +78,7 @@
 PRIVATE
     Qt5::Network        #QNetworkAccessManager in kaboutapplicationpersonmodel_p
     Qt5::PrintSupport   #QPrinter in kshortcutseditor
+    KF5::CoreAddons     #KAboutData
     KF5::WidgetsAddons
     KF5::ItemViews           #KWidgetItemDelegate in KAboutApplicationPersonListDelegate
     KF5::I18n                #i18n and i18nc in many places
diff -urw kxmlgui-5.12.0/tests/CMakeLists.txt XmlGui/tests/CMakeLists.txt
--- kxmlgui-5.12.0/tests/CMakeLists.txt	2015-07-05 12:16:23.000000000 +0300
+++ XmlGui/tests/CMakeLists.txt	2015-07-30 20:57:54.448450233 +0300
@@ -6,7 +6,7 @@
 macro(xmlgui_executable_tests)
   foreach(_testname ${ARGN})
     add_executable(${_testname} ${_testname}.cpp)
-    target_link_libraries(${_testname} Qt5::Test KF5::WidgetsAddons KF5::I18n KF5::XmlGui)
+    target_link_libraries(${_testname} Qt5::Test KF5::CoreAddons KF5::WidgetsAddons KF5::I18n KF5::XmlGui)
     ecm_mark_as_test(${_testname})
   endforeach(_testname)
 endmacro()
