diff -urw kiconthemes-5.12.0/CMakeLists.txt IconThemes/CMakeLists.txt
--- kiconthemes-5.12.0/CMakeLists.txt	2015-07-05 12:05:08.000000000 +0300
+++ IconThemes/CMakeLists.txt	2015-08-14 21:35:33.214578159 +0300
@@ -35,6 +35,7 @@
 find_package(Qt5DBus ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
 
 find_package(KF5I18n ${KF5_DEP_VERSION} REQUIRED)
+find_package(KF5CoreAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5ConfigWidgets ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5WidgetsAddons ${KF5_DEP_VERSION} REQUIRED)
 find_package(KF5ItemViews ${KF5_DEP_VERSION} REQUIRED)
diff -urw kiconthemes-5.12.0/src/CMakeLists.txt IconThemes/src/CMakeLists.txt
--- kiconthemes-5.12.0/src/CMakeLists.txt	2015-07-05 12:05:08.000000000 +0300
+++ IconThemes/src/CMakeLists.txt	2015-08-14 21:37:23.371991267 +0300
@@ -29,6 +29,7 @@
     KF5::WidgetsAddons # for KPixmapSequence family
     KF5::ItemViews      # for KListWidgetSearchLine
     KF5::ConfigWidgets # for KColorScheme
+    KF5::CoreAddons    # for kshareddatacache.h
 )
 
 set_target_properties(KF5IconThemes PROPERTIES VERSION   ${KICONTHEMES_VERSION_STRING}
