diff -Naur x2/CMakeLists.txt x/CMakeLists.txt
--- x2/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ x/CMakeLists.txt	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,127 @@
+# This script compiles update-mime-database on windows
+# needed libs:
+# libxml2 >= 2.6
+# libintl >= 0.14
+# iconv
+
+# additionally you'll need the following sources
+# shared-mimeinfo
+# kdewin32
+# glib >= 2.12
+# extract all into on directory and adjust the following paths:
+
+project(update-mime-database)
+cmake_minimum_required(VERSION 2.6.2)
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}")
+
+if(COMMAND cmake_policy)
+  cmake_policy(SET CMP0003 NEW)
+endif(COMMAND cmake_policy)
+
+include(CheckIncludeFiles)
+include(CheckSymbolExists)
+
+if(MINGW)
+	# set MINGW32 or MINGW64
+	INCLUDE(CheckMingwVersion)
+endif(MINGW)
+
+set(GLIB_DIR     "" CACHE FILEPATH "Path to glib sources")
+
+# libxml2
+find_package(LIBXML2 REQUIRED)
+# libintl
+find_package(Libintl REQUIRED)
+
+include_directories(
+   ${GLIB_DIR}
+   ${GLIB_DIR}/glib
+   ${LIBINTL_INCLUDE_DIR}
+   ${LIBXML2_INCLUDE_DIR}
+   ${CMAKE_BINARY_DIR}
+)
+
+if(MSVC)
+    include_directories(
+        ${CMAKE_SOURCE_DIR}/headers
+    )
+endif(MSVC)
+
+# configure checks
+check_include_files(inttypes.h  HAVE_INTTYPES_H)
+check_include_files(locale.h    HAVE_LOCALE_H)
+check_include_files(memory.h    HAVE_MEMORY_H)
+check_include_files(stdint.h    HAVE_STDINT_H)
+check_include_files(stdlib.h    HAVE_STDLIB_H)
+check_include_files(strings.h   HAVE_STRINGS_H)
+check_include_files(string.h    HAVE_STRING_H)
+check_include_files(sys/stat.h  HAVE_SYS_STAT_H)
+check_include_files(sys/types.h HAVE_SYS_TYPES_H)
+check_include_files(unistd.h    HAVE_UNISTD_H)
+check_symbol_exists(LC_MESSAGES locale.h HAVE_LC_MESSAGES)
+
+set(PACKAGE           "update-mime-database")
+set(PACKAGE_BUGREPORT "https://bugs.freedesktop.org/enter_bug.cgi?product=shared-mime-info")
+set(PACKAGE_NAME      "shared-mime-info")
+set(PACKAGE_STRING    "shared-mime-info 0.71")
+set(PACKAGE_TARNAME   "shared-mime-info")
+set(PACKAGE_VERSION   "0.71")
+set(VERSION           "0.71")
+configure_file(${CMAKE_SOURCE_DIR}/config.h.cmake ${CMAKE_BINARY_DIR}/config.h)
+
+set(GLIB_SOURCES
+    ${GLIB_DIR}/glib/galiasdef.c
+    ${GLIB_DIR}/glib/gatomic.c
+    ${GLIB_DIR}/glib/garray.c
+    ${GLIB_DIR}/glib/gbitlock.c
+    ${GLIB_DIR}/glib/gconvert.c
+    ${GLIB_DIR}/glib/gdataset.c
+    ${GLIB_DIR}/glib/gerror.c
+    ${GLIB_DIR}/glib/gfileutils.c
+    ${GLIB_DIR}/glib/ghash.c
+    ${GLIB_DIR}/glib/glist.c
+    ${GLIB_DIR}/glib/gmain.c
+    ${GLIB_DIR}/glib/gmem.c
+    ${GLIB_DIR}/glib/gmessages.c
+    ${GLIB_DIR}/glib/gpoll.c
+    ${GLIB_DIR}/glib/gprimes.c
+    ${GLIB_DIR}/glib/gprintf.c
+    ${GLIB_DIR}/glib/gqsort.c
+    ${GLIB_DIR}/glib/gqueue.c
+    ${GLIB_DIR}/glib/grand.c
+    ${GLIB_DIR}/glib/gutf8.c
+    ${GLIB_DIR}/glib/gutils.c
+    ${GLIB_DIR}/glib/gstdio.c
+    ${GLIB_DIR}/glib/gslice.c
+    ${GLIB_DIR}/glib/gslist.c
+    ${GLIB_DIR}/glib/gstrfuncs.c
+    ${GLIB_DIR}/glib/gstring.c
+    ${GLIB_DIR}/glib/gtestutils.c
+    ${GLIB_DIR}/glib/gthread.c
+    ${GLIB_DIR}/glib/gtimer.c
+    ${GLIB_DIR}/glib/guniprop.c
+    ${GLIB_DIR}/glib/gunidecomp.c
+    ${GLIB_DIR}/glib/gwin32.c
+    ${GLIB_DIR}/glib/gnulib/printf.c
+    ${GLIB_DIR}/glib/gnulib/printf-args.c
+    ${GLIB_DIR}/glib/gnulib/printf-parse.c
+    ${GLIB_DIR}/glib/gnulib/vasnprintf.c
+    ${GLIB_DIR}/glib/libcharset/localcharset.c
+)
+
+if(MSVC)
+    add_definitions( -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -wd4018)
+    set(additional_SOURCES dirent.c unistd.c)
+endif(MSVC)
+
+add_definitions(-DGLIB_VAR= -DWINPOSIX_EXPORT= -DKDEWIN_EXPORT= -DG_DISABLE_DEPRECATED -DLIBDIR=\"lib\")
+
+add_executable(update-mime-database update-mime-database.c ${GLIB_SOURCES} ${additional_SOURCES})
+
+target_link_libraries(update-mime-database
+        ${LIBINTL_LIBRARIES}
+        ${LIBXML2_LIBRARIES}
+)
+
+install(TARGETS update-mime-database RUNTIME DESTINATION bin)
+install(FILES ${CMAKE_SOURCE_DIR}/freedesktop.org.xml DESTINATION share/mime/packages)
diff -Naur x2/FindLibintl.cmake x/FindLibintl.cmake
--- x2/FindLibintl.cmake	1970-01-01 01:00:00.000000000 +0100
+++ x/FindLibintl.cmake	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,46 @@
+# Try to find Libintl functionality
+# Once done this will define
+#
+#  LIBINTL_FOUND - system has Libintl
+#  LIBINTL_INCLUDE_DIR - Libintl include directory
+#  LIBINTL_LIBRARIES - Libraries needed to use Libintl
+#
+# TODO: This will enable translations only if Gettext functionality is
+# present in libc. Must have more robust system for release, where Gettext
+# functionality can also reside in standalone Gettext library, or the one
+# embedded within kdelibs (cf. gettext.m4 from Gettext source).
+#
+# Copyright (c) 2006, Chusslove Illich, <caslav.ilic@gmx.net>
+# Copyright (c) 2007, Alexander Neundorf, <neundorf@kde.org>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+if(LIBINTL_INCLUDE_DIR AND LIBINTL_LIB_FOUND)
+  set(Libintl_FIND_QUIETLY TRUE)
+endif(LIBINTL_INCLUDE_DIR AND LIBINTL_LIB_FOUND)
+
+find_path(LIBINTL_INCLUDE_DIR libintl.h)
+
+set(LIBINTL_LIB_FOUND FALSE)
+
+if(LIBINTL_INCLUDE_DIR)
+  include(CheckFunctionExists)
+  check_function_exists(dgettext LIBINTL_LIBC_HAS_DGETTEXT)
+
+  if (LIBINTL_LIBC_HAS_DGETTEXT)
+    set(LIBINTL_LIBRARIES)
+    set(LIBINTL_LIB_FOUND TRUE CACHE BOOL "Libintl found" FORCE)
+  else (LIBINTL_LIBC_HAS_DGETTEXT)
+    find_library(LIBINTL_LIBRARIES NAMES intl libintl )
+    if(LIBINTL_LIBRARIES)
+      set(LIBINTL_LIB_FOUND TRUE CACHE BOOL "Libintl found" FORCE)
+    endif(LIBINTL_LIBRARIES)
+  endif (LIBINTL_LIBC_HAS_DGETTEXT)
+
+endif(LIBINTL_INCLUDE_DIR)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Libintl  DEFAULT_MSG  LIBINTL_INCLUDE_DIR  LIBINTL_LIB_FOUND)
+
+mark_as_advanced(LIBINTL_INCLUDE_DIR LIBINTL_LIBRARIES LIBINTL_LIBC_HAS_DGETTEXT)
diff -Naur x2/config.h.cmake x/config.h.cmake
--- x2/config.h.cmake	1970-01-01 01:00:00.000000000 +0100
+++ x/config.h.cmake	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,71 @@
+/* always defined to indicate that i18n is enabled */
+#cmakedefine ENABLE_NLS 1
+
+/* Define to 1 if you have the `bind_textdomain_codeset' function. */
+#cmakedefine HAVE_BIND_TEXTDOMAIN_CODESET 1
+
+/* Define to 1 if you have the `dcgettext' function. */
+#cmakedefine HAVE_DCGETTEXT 1
+
+/* Define if the GNU gettext() function is already present or preinstalled. */
+#cmakedefine HAVE_GETTEXT 1
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#cmakedefine HAVE_INTTYPES_H 1
+
+/* Define if your <locale.h> file defines LC_MESSAGES. */
+#cmakedefine HAVE_LC_MESSAGES 1
+
+/* Define to 1 if you have the <locale.h> header file. */
+#cmakedefine HAVE_LOCALE_H 1
+
+/* Define to 1 if you have the <memory.h> header file. */
+#cmakedefine HAVE_MEMORY_H 1
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#cmakedefine HAVE_STDINT_H 1
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#cmakedefine HAVE_STDLIB_H 1
+
+/* Define to 1 if you have the <strings.h> header file. */
+#cmakedefine HAVE_STRINGS_H 1
+
+/* Define to 1 if you have the <string.h> header file. */
+#cmakedefine HAVE_STRING_H 1
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#cmakedefine HAVE_SYS_STAT_H 1
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#cmakedefine HAVE_SYS_TYPES_H 1
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#cmakedefine HAVE_UNISTD_H 1
+
+/* Name of package */
+#cmakedefine PACKAGE "@PACKAGE@"
+
+/* Define to the address where bug reports for this package should be sent. */
+#cmakedefine PACKAGE_BUGREPORT "@PACKAGE_BUGREPORT@"
+
+/* Define to the full name of this package. */
+#cmakedefine PACKAGE_NAME "@PACKAGE_NAME@"
+
+/* Define to the full name and version of this package. */
+#cmakedefine PACKAGE_STRING "@PACKAGE_STRING@"
+
+/* Define to the one symbol short name of this package. */
+#cmakedefine PACKAGE_TARNAME "@PACKAGE_TARNAME@"
+
+/* Define to the version of this package. */
+#cmakedefine PACKAGE_VERSION "@PACKAGE_VERSION@"
+
+/* Define to 1 if you have the ANSI C header files. */
+#cmakedefine STDC_HEADERS 1
+
+/* Version number of package */
+#cmakedefine VERSION  "@VERSION@"
+
+/* Define to empty if `const' does not conform to ANSI C. */
+// #define const
diff -Naur x2/dirent.c x/dirent.c
--- x2/dirent.c	1970-01-01 01:00:00.000000000 +0100
+++ x/dirent.c	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,149 @@
+/* ====================================================================
+ * The Apache Software License, Version 1.1
+ *
+ * Copyright (c) 2000-2002 The Apache Software Foundation.  All rights
+ * reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. The end-user documentation included with the redistribution,
+ *    if any, must include the following acknowledgment:
+ *       "This product includes software developed by the
+ *        Apache Software Foundation (http://www.apache.org/)."
+ *    Alternately, this acknowledgment may appear in the software itself,
+ *    if and wherever such third-party acknowledgments normally appear.
+ *
+ * 4. The names "Apache" and "Apache Software Foundation" must
+ *    not be used to endorse or promote products derived from this
+ *    software without prior written permission. For written
+ *    permission, please contact apache@apache.org.
+ *
+ * 5. Products derived from this software may not be called "Apache",
+ *    nor may "Apache" appear in their name, without prior written
+ *    permission of the Apache Software Foundation.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE APACHE SOFTWARE FOUNDATION OR
+ * ITS CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ * USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ * Portions of this software are based upon public domain software
+ * originally written at the National Center for Supercomputing Applications,
+ * University of Illinois, Urbana-Champaign.
+ */
+
+#include <windows.h>
+
+#include <malloc.h>
+#include <string.h>
+#include <errno.h>
+
+#include "dirent.h"
+
+/**********************************************************************
+ * Implement dirent-style opendir/readdir/closedir on Window 95/NT
+ *
+ * Functions defined are opendir(), readdir() and closedir() with the
+ * same prototypes as the normal dirent.h implementation.
+ *
+ * Does not implement telldir(), seekdir(), rewinddir() or scandir(). 
+ * The dirent struct is compatible with Unix, except that d_ino is 
+ * always 1 and d_off is made up as we go along.
+ *
+ * The DIR typedef is not compatible with Unix.
+ **********************************************************************/
+
+#ifndef __MINGW32__
+
+DIR * opendir(const char *dir)
+{
+    DIR *dp;
+    char *filespec;
+    long handle;
+    int index;
+
+    filespec = malloc(strlen(dir) + 2 + 1);
+    strcpy(filespec, dir);
+    index = strlen(filespec) - 1;
+    if (index >= 0 && (filespec[index] == '/' || filespec[index] == '\\'))
+        filespec[index] = '\0';
+    strcat(filespec, "\\*");
+
+    dp = (DIR *)malloc(sizeof(DIR));
+    dp->offset = 0;
+    dp->finished = 0;
+    dp->dir = strdup(dir);
+
+    if ((handle = _findfirst(filespec, &(dp->fileinfo))) < 0) {
+        if (errno == ENOENT)
+            dp->finished = 1;
+        else
+        return NULL;
+    }
+
+    dp->handle = handle;
+    free(filespec);
+
+    return dp;
+}
+
+struct dirent * readdir(DIR *dp)
+{
+    if (!dp || dp->finished) return NULL;
+
+    if (dp->offset != 0) {
+        if (_findnext(dp->handle, &(dp->fileinfo)) < 0) {
+            dp->finished = 1;
+            /* posix does not set errno in this case */
+            errno = 0;
+            return NULL;
+        }
+    }
+    dp->offset++;
+
+    strncpy(dp->dent.d_name, dp->fileinfo.name, _MAX_FNAME);
+#ifdef KDEWIN_HAVE_DIRENT_D_TYPE
+    dp->dent.d_type = DT_UNKNOWN;
+#endif    
+    dp->dent.d_ino = 1;
+    dp->dent.d_reclen = strlen(dp->dent.d_name);
+    dp->dent.d_off = dp->offset;
+
+    return &(dp->dent);
+}
+
+int closedir(DIR *dp)
+{
+    if (!dp) return 0;
+    if ((HANDLE)dp->handle != INVALID_HANDLE_VALUE) _findclose(dp->handle);
+    if (dp->dir) free(dp->dir);
+    if (dp) free(dp);
+
+    return 0;
+}
+
+#endif // #ifndef __MINGW32__
diff -Naur x2/dirent.h x/dirent.h
--- x2/dirent.h	1970-01-01 01:00:00.000000000 +0100
+++ x/dirent.h	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,88 @@
+/* This file is part of the KDE project
+   Copyright (C) 2000 Werner Almesberger
+
+   libc/sys/linux/sys/dirent.h - Directory entry as returned by readdir
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public License
+   along with this program; see the file COPYING.  If not, write to
+   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110-1301, USA.
+*/
+
+#ifndef SMI_DIRENT_H
+#define SMI_DIRENT_H
+
+#include <io.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <sys/types.h>
+
+//#ifdef HAVE_CONFIG_H
+//#include "config.h"
+//#endif
+//#ifdef HAVE_STDINT_H
+//#include <stdint.h>
+//#else
+//typedef off_t long int
+//#endif
+
+#define KDEWIN_HAVE_DIRENT_D_TYPE
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define HAVE_NO_D_NAMLEN	/* no struct dirent->d_namlen */
+#define HAVE_DD_LOCK  		/* have locking mechanism */
+
+#define MAXNAMLEN 255		/* sizeof(struct dirent.d_name)-1 */
+
+#define __dirfd(dir) (dir)->dd_fd
+
+enum
+  {
+    DT_UNKNOWN = 0,
+    DT_FIFO = 1,
+    DT_CHR = 2,
+    DT_DIR = 4,
+    DT_BLK = 6,
+    DT_REG = 8,
+    DT_LNK = 10,
+    DT_SOCK = 12,
+    DT_WHT = 14
+  };
+
+/* struct dirent - same as Unix */
+struct dirent {
+    long d_ino;                    /* inode (always 1 in WIN32) */
+    off_t d_off;                /* offset to this dirent */
+    unsigned short d_reclen;    /* length of d_name */
+    unsigned char d_type;
+    char d_name[_MAX_FNAME+1];    /* filename (null terminated) */
+};
+
+/* typedef DIR - not the same as Unix */
+typedef struct {
+    long handle;                /* _findfirst/_findnext handle */
+    short offset;                /* offset into directory */
+    short finished;             /* 1 if there are not more files */
+    struct _finddata_t fileinfo;  /* from _findfirst/_findnext */
+    char *dir;                  /* the dir we are reading */
+    struct dirent dent;         /* the dirent to return */
+} DIR;
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif  // KDEWIN_DIRENT_H
diff -Naur x2/glib-2.24.0/glib/galias.h x/glib-2.24.0/glib/galias.h
--- x2/glib-2.24.0/glib/galias.h	2010-03-26 17:48:20.000000000 +0100
+++ x/glib-2.24.0/glib/galias.h	2013-04-12 12:33:22.000000000 +0200
@@ -2,7 +2,7 @@
 
 #ifndef DISABLE_VISIBILITY
 
-#include "glibconfig.h"
+#include "glibconfig.h.win32"
 
 #ifdef G_HAVE_GNUC_VISIBILITY
 
diff -Naur x2/glib-2.24.0/glib/galiasdef.c x/glib-2.24.0/glib/galiasdef.c
--- x2/glib-2.24.0/glib/galiasdef.c	2010-03-26 17:48:20.000000000 +0100
+++ x/glib-2.24.0/glib/galiasdef.c	2013-04-12 12:32:22.000000000 +0200
@@ -2,7 +2,7 @@
 
 #ifndef DISABLE_VISIBILITY
 
-#include "glibconfig.h"
+#include "glibconfig.h.win32"
 
 #ifdef G_HAVE_GNUC_VISIBILITY
 
diff -Naur x2/glib-2.24.0/glib/garray.c x/glib-2.24.0/glib/garray.c
--- x2/glib-2.24.0/glib/garray.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/garray.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 #include <stdlib.h>
diff -Naur x2/glib-2.24.0/glib/gasyncqueue.c x/glib-2.24.0/glib/gasyncqueue.c
--- x2/glib-2.24.0/glib/gasyncqueue.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gasyncqueue.c	2013-04-12 12:32:22.000000000 +0200
@@ -24,7 +24,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gatomic-gcc.c x/glib-2.24.0/glib/gatomic-gcc.c
--- x2/glib-2.24.0/glib/gatomic-gcc.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gatomic-gcc.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gatomic.c x/glib-2.24.0/glib/gatomic.c
--- x2/glib-2.24.0/glib/gatomic.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gatomic.c	2013-04-12 12:32:22.000000000 +0200
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #if defined (G_ATOMIC_ARM)
 #include <sched.h>
diff -Naur x2/glib-2.24.0/glib/gbacktrace.c x/glib-2.24.0/glib/gbacktrace.c
--- x2/glib-2.24.0/glib/gbacktrace.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gbacktrace.c	2013-04-12 12:32:22.000000000 +0200
@@ -29,7 +29,7 @@
  * then
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <signal.h>
 #include <stdarg.h>
diff -Naur x2/glib-2.24.0/glib/gbase64.c x/glib-2.24.0/glib/gbase64.c
--- x2/glib-2.24.0/glib/gbase64.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gbase64.c	2013-04-12 12:32:22.000000000 +0200
@@ -23,7 +23,7 @@
  *    Jeffrey Stedfast <fejj@ximian.com>
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
diff -Naur x2/glib-2.24.0/glib/gbitlock.c x/glib-2.24.0/glib/gbitlock.c
--- x2/glib-2.24.0/glib/gbitlock.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gbitlock.c	2013-04-12 12:32:22.000000000 +0200
@@ -27,7 +27,7 @@
 #include <glib/gthread.h>
 
 #include "gthreadprivate.h"
-#include "config.h"
+#include "config.h.win32"
 
 #include "galias.h"
 
diff -Naur x2/glib-2.24.0/glib/gbookmarkfile.c x/glib-2.24.0/glib/gbookmarkfile.c
--- x2/glib-2.24.0/glib/gbookmarkfile.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gbookmarkfile.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
  * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gbookmarkfile.h"
 
diff -Naur x2/glib-2.24.0/glib/gbuffer.c x/glib-2.24.0/glib/gbuffer.c
--- x2/glib-2.24.0/glib/gbuffer.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gbuffer.c	2013-04-12 12:32:22.000000000 +0200
@@ -19,7 +19,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gbuffer.h"
 
diff -Naur x2/glib-2.24.0/glib/gcache.c x/glib-2.24.0/glib/gcache.c
--- x2/glib-2.24.0/glib/gcache.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gcache.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gchecksum.c x/glib-2.24.0/glib/gchecksum.c
--- x2/glib-2.24.0/glib/gchecksum.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gchecksum.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,11 +18,11 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
-#include "glibconfig.h"
+#include "glibconfig.h.win32"
 #include "gchecksum.h"
 #include "glib.h"
 #include "glibintl.h"
diff -Naur x2/glib-2.24.0/glib/gcompletion.c x/glib-2.24.0/glib/gcompletion.c
--- x2/glib-2.24.0/glib/gcompletion.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gcompletion.c	2013-04-12 12:32:23.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
diff -Naur x2/glib-2.24.0/glib/gconvert.c x/glib-2.24.0/glib/gconvert.c
--- x2/glib-2.24.0/glib/gconvert.c	2010-03-17 21:56:26.000000000 +0100
+++ x/glib-2.24.0/glib/gconvert.c	2013-04-12 12:32:22.000000000 +0200
@@ -20,7 +20,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 
diff -Naur x2/glib-2.24.0/glib/gdataset.c x/glib-2.24.0/glib/gdataset.c
--- x2/glib-2.24.0/glib/gdataset.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gdataset.c	2013-04-12 12:32:22.000000000 +0200
@@ -30,7 +30,7 @@
  * MT safe ; except for g_data*_foreach()
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
diff -Naur x2/glib-2.24.0/glib/gdate.c x/glib-2.24.0/glib/gdate.c
--- x2/glib-2.24.0/glib/gdate.c	2010-02-11 07:34:48.000000000 +0100
+++ x/glib-2.24.0/glib/gdate.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #define DEBUG_MSG(x)	/* */
 #ifdef G_ENABLE_DEBUG
diff -Naur x2/glib-2.24.0/glib/gdir.c x/glib-2.24.0/glib/gdir.c
--- x2/glib-2.24.0/glib/gdir.c	2009-08-28 05:51:21.000000000 +0200
+++ x/glib-2.24.0/glib/gdir.c	2013-04-12 12:32:23.000000000 +0200
@@ -22,7 +22,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <errno.h>
 #include <string.h>
diff -Naur x2/glib-2.24.0/glib/gerror.c x/glib-2.24.0/glib/gerror.c
--- x2/glib-2.24.0/glib/gerror.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/gerror.c	2013-04-12 12:32:22.000000000 +0200
@@ -24,7 +24,7 @@
  * GLib at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gfileutils.c x/glib-2.24.0/glib/gfileutils.c
--- x2/glib-2.24.0/glib/gfileutils.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gfileutils.c	2013-04-12 12:32:23.000000000 +0200
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 
diff -Naur x2/glib-2.24.0/glib/ghash.c x/glib-2.24.0/glib/ghash.c
--- x2/glib-2.24.0/glib/ghash.c	2010-03-26 13:34:20.000000000 +0100
+++ x/glib-2.24.0/glib/ghash.c	2013-04-12 12:32:23.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>  /* memset */
 
diff -Naur x2/glib-2.24.0/glib/ghook.c x/glib-2.24.0/glib/ghook.c
--- x2/glib-2.24.0/glib/ghook.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/ghook.c	2013-04-12 12:32:23.000000000 +0200
@@ -31,7 +31,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/ghostutils.c x/glib-2.24.0/glib/ghostutils.c
--- x2/glib-2.24.0/glib/ghostutils.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/ghostutils.c	2013-04-12 12:32:23.000000000 +0200
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "glibintl.h"
diff -Naur x2/glib-2.24.0/glib/gi18n-lib.h x/glib-2.24.0/glib/gi18n-lib.h
--- x2/glib-2.24.0/glib/gi18n-lib.h	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gi18n-lib.h	2013-04-12 12:33:22.000000000 +0200
@@ -26,7 +26,7 @@
 #include <string.h>
 
 #ifndef GETTEXT_PACKAGE
-#error You must define GETTEXT_PACKAGE before including gi18n-lib.h.  Did you forget to include config.h?
+#error You must define GETTEXT_PACKAGE before including gi18n-lib.h.  Did you forget to include config.h.win32?
 #endif
 
 #define  _(String) ((char *) g_dgettext (GETTEXT_PACKAGE, String))
diff -Naur x2/glib-2.24.0/glib/giochannel.c x/glib-2.24.0/glib/giochannel.c
--- x2/glib-2.24.0/glib/giochannel.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/giochannel.c	2013-04-12 12:32:23.000000000 +0200
@@ -31,7 +31,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 #include <errno.h>
diff -Naur x2/glib-2.24.0/glib/giounix.c x/glib-2.24.0/glib/giounix.c
--- x2/glib-2.24.0/glib/giounix.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/giounix.c	2013-04-12 12:32:22.000000000 +0200
@@ -31,7 +31,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #define _POSIX_SOURCE		/* for SSIZE_MAX */
 
diff -Naur x2/glib-2.24.0/glib/giowin32.c x/glib-2.24.0/glib/giowin32.c
--- x2/glib-2.24.0/glib/giowin32.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/giowin32.c	2013-04-12 12:32:22.000000000 +0200
@@ -67,7 +67,7 @@
  * above bugs continue to work.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 
diff -Naur x2/glib-2.24.0/glib/gkeyfile.c x/glib-2.24.0/glib/gkeyfile.c
--- x2/glib-2.24.0/glib/gkeyfile.c	2010-03-13 20:05:53.000000000 +0100
+++ x/glib-2.24.0/glib/gkeyfile.c	2013-04-12 12:32:22.000000000 +0200
@@ -21,7 +21,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gkeyfile.h"
 
diff -Naur x2/glib-2.24.0/glib/glibintl.h x/glib-2.24.0/glib/glibintl.h
--- x2/glib-2.24.0/glib/glibintl.h	2009-04-02 03:34:11.000000000 +0200
+++ x/glib-2.24.0/glib/glibintl.h	2013-04-12 12:33:22.000000000 +0200
@@ -2,7 +2,7 @@
 #define __GLIBINTL_H__
 
 #ifndef SIZEOF_CHAR
-#error "config.h must be included prior to glibintl.h"
+#error "config.h.win32 must be included prior to glibintl.h"
 #endif
 
 G_CONST_RETURN gchar *glib_gettext (const gchar *str) G_GNUC_FORMAT(1);
diff -Naur x2/glib-2.24.0/glib/glist.c x/glib-2.24.0/glib/glist.c
--- x2/glib-2.24.0/glib/glist.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/glist.c	2013-04-12 12:32:23.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gmacros.h x/glib-2.24.0/glib/gmacros.h
--- x2/glib-2.24.0/glib/gmacros.h	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gmacros.h	2013-04-12 12:33:22.000000000 +0200
@@ -25,7 +25,7 @@
  */
 
 /* This file must not include any other glib header file and must thus
- * not refer to variables from glibconfig.h
+ * not refer to variables from glibconfig.h.win32
  */
 
 #if defined(G_DISABLE_SINGLE_INCLUDES) && !defined (__GLIB_H_INSIDE__) && !defined (GLIB_COMPILATION)
diff -Naur x2/glib-2.24.0/glib/gmain.c x/glib-2.24.0/glib/gmain.c
--- x2/glib-2.24.0/glib/gmain.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/gmain.c	2013-04-12 12:32:22.000000000 +0200
@@ -31,7 +31,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 /* Uncomment the next line (and the corresponding line in gpoll.c) to
  * enable debugging printouts if the environment variable
diff -Naur x2/glib-2.24.0/glib/gmappedfile.c x/glib-2.24.0/glib/gmappedfile.c
--- x2/glib-2.24.0/glib/gmappedfile.c	2010-03-17 21:55:41.000000000 +0100
+++ x/glib-2.24.0/glib/gmappedfile.c	2013-04-12 12:32:23.000000000 +0200
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <errno.h>
 #include <sys/types.h> 
@@ -32,7 +32,7 @@
 #include <sys/mman.h>
 #endif
 
-#include "glibconfig.h"
+#include "glibconfig.h.win32"
 
 #ifdef G_OS_WIN32
 #include <windows.h>
diff -Naur x2/glib-2.24.0/glib/gmarkup.c x/glib-2.24.0/glib/gmarkup.c
--- x2/glib-2.24.0/glib/gmarkup.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gmarkup.c	2013-04-12 12:32:22.000000000 +0200
@@ -19,7 +19,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdarg.h>
 #include <string.h>
diff -Naur x2/glib-2.24.0/glib/gmem.c x/glib-2.24.0/glib/gmem.c
--- x2/glib-2.24.0/glib/gmem.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gmem.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdlib.h>
 #include <string.h>
diff -Naur x2/glib-2.24.0/glib/gmessages.c x/glib-2.24.0/glib/gmessages.c
--- x2/glib-2.24.0/glib/gmessages.c	2009-09-30 02:45:21.000000000 +0200
+++ x/glib-2.24.0/glib/gmessages.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdlib.h>
 #include <stdarg.h>
diff -Naur x2/glib-2.24.0/glib/gnode.c x/glib-2.24.0/glib/gnode.c
--- x2/glib-2.24.0/glib/gnode.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gnode.c	2013-04-12 12:32:22.000000000 +0200
@@ -31,7 +31,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gnulib/asnprintf.c x/glib-2.24.0/glib/gnulib/asnprintf.c
--- x2/glib-2.24.0/glib/gnulib/asnprintf.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gnulib/asnprintf.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
    USA.  */
 
 #ifdef HAVE_CONFIG_H
-# include <config.h>
+# include <config.h.win32>
 #endif
 
 #include "g-gnulib.h"
diff -Naur x2/glib-2.24.0/glib/gnulib/g-gnulib.h x/glib-2.24.0/glib/gnulib/g-gnulib.h
--- x2/glib-2.24.0/glib/gnulib/g-gnulib.h	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gnulib/g-gnulib.h	2013-04-12 12:33:22.000000000 +0200
@@ -18,7 +18,7 @@
  */
 #ifndef __G_GNULIB_H__
 
-#include "config.h"
+#include "config.h.win32"
 #include <stdlib.h>
 #include "glib/glib.h"
 #include "glib/galias.h"
diff -Naur x2/glib-2.24.0/glib/gnulib/printf-args.c x/glib-2.24.0/glib/gnulib/printf-args.c
--- x2/glib-2.24.0/glib/gnulib/printf-args.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gnulib/printf-args.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
    USA.  */
 
 #ifdef HAVE_CONFIG_H
-# include <config.h>
+# include <config.h.win32>
 #endif
 
 #include "g-gnulib.h"
diff -Naur x2/glib-2.24.0/glib/gnulib/printf-parse.c x/glib-2.24.0/glib/gnulib/printf-parse.c
--- x2/glib-2.24.0/glib/gnulib/printf-parse.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gnulib/printf-parse.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
    USA.  */
 
 #ifdef HAVE_CONFIG_H
-# include <config.h>
+# include <config.h.win32>
 #endif
 
 #include "g-gnulib.h"
diff -Naur x2/glib-2.24.0/glib/gnulib/printf.c x/glib-2.24.0/glib/gnulib/printf.c
--- x2/glib-2.24.0/glib/gnulib/printf.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gnulib/printf.c	2013-04-12 12:32:22.000000000 +0200
@@ -25,7 +25,7 @@
  */
 
 #ifdef HAVE_CONFIG_H
-# include <config.h>
+# include <config.h.win32>
 #endif
 #include <string.h>
 #include <stdlib.h>
diff -Naur x2/glib-2.24.0/glib/gnulib/vasnprintf.c x/glib-2.24.0/glib/gnulib/vasnprintf.c
--- x2/glib-2.24.0/glib/gnulib/vasnprintf.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gnulib/vasnprintf.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,7 +18,7 @@
 
 #ifndef _WIN32
 /* Tell glibc's <stdio.h> to provide a prototype for snprintf().
-   This must come before <config.h> because <config.h> may include
+   This must come before <config.h.win32> because <config.h> may include
    <features.h>, and once <features.h> has been included, it's too late.  */
 #ifndef _GNU_SOURCE
 # define _GNU_SOURCE    1
@@ -26,7 +26,7 @@
 #endif
 
 #ifdef HAVE_CONFIG_H
-# include <config.h>
+# include <config.h.win32>
 #endif
 #include "glib/galloca.h"
 
diff -Naur x2/glib-2.24.0/glib/goption.c x/glib-2.24.0/glib/goption.c
--- x2/glib-2.24.0/glib/goption.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/goption.c	2013-04-12 12:32:22.000000000 +0200
@@ -125,7 +125,7 @@
  * </programlisting></informalexample>
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "goption.h"
 #include "glib.h"
diff -Naur x2/glib-2.24.0/glib/gpattern.c x/glib-2.24.0/glib/gpattern.c
--- x2/glib-2.24.0/glib/gpattern.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gpattern.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
diff -Naur x2/glib-2.24.0/glib/gpoll.c x/glib-2.24.0/glib/gpoll.c
--- x2/glib-2.24.0/glib/gpoll.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gpoll.c	2013-04-12 12:32:22.000000000 +0200
@@ -32,7 +32,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 /* Uncomment the next line (and the corresponding line in gmain.c) to
  * enable debugging printouts if the environment variable
diff -Naur x2/glib-2.24.0/glib/gprimes.c x/glib-2.24.0/glib/gprimes.c
--- x2/glib-2.24.0/glib/gprimes.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gprimes.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gprintf.c x/glib-2.24.0/glib/gprintf.c
--- x2/glib-2.24.0/glib/gprintf.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gprintf.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #ifndef _WIN32
 #define _GNU_SOURCE		/* For vasprintf */
diff -Naur x2/glib-2.24.0/glib/gqsort.c x/glib-2.24.0/glib/gqsort.c
--- x2/glib-2.24.0/glib/gqsort.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gqsort.c	2013-04-12 12:32:22.000000000 +0200
@@ -32,7 +32,7 @@
  * at ftp://ftp.gtk.org/pub/gtk/.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <limits.h>
 #include <stdlib.h>
diff -Naur x2/glib-2.24.0/glib/gqueue.c x/glib-2.24.0/glib/gqueue.c
--- x2/glib-2.24.0/glib/gqueue.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gqueue.c	2013-04-12 12:32:22.000000000 +0200
@@ -24,7 +24,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/grand.c x/glib-2.24.0/glib/grand.c
--- x2/glib-2.24.0/glib/grand.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/grand.c	2013-04-12 12:32:22.000000000 +0200
@@ -36,7 +36,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <math.h>
 #include <errno.h>
diff -Naur x2/glib-2.24.0/glib/gregex.c x/glib-2.24.0/glib/gregex.c
--- x2/glib-2.24.0/glib/gregex.c	2009-09-19 01:20:05.000000000 +0200
+++ x/glib-2.24.0/glib/gregex.c	2013-04-12 12:32:22.000000000 +0200
@@ -19,7 +19,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
diff -Naur x2/glib-2.24.0/glib/grel.c x/glib-2.24.0/glib/grel.c
--- x2/glib-2.24.0/glib/grel.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/grel.c	2013-04-12 12:32:22.000000000 +0200
@@ -27,7 +27,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdarg.h>
 #include <string.h>
diff -Naur x2/glib-2.24.0/glib/gscanner.c x/glib-2.24.0/glib/gscanner.c
--- x2/glib-2.24.0/glib/gscanner.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gscanner.c	2013-04-12 12:32:22.000000000 +0200
@@ -31,7 +31,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include	<errno.h>
 #include	<stdlib.h>
diff -Naur x2/glib-2.24.0/glib/gsequence.c x/glib-2.24.0/glib/gsequence.c
--- x2/glib-2.24.0/glib/gsequence.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gsequence.c	2013-04-12 12:32:23.000000000 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gshell.c x/glib-2.24.0/glib/gshell.c
--- x2/glib-2.24.0/glib/gshell.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gshell.c	2013-04-12 12:32:22.000000000 +0200
@@ -20,7 +20,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <string.h>
 
diff -Naur x2/glib-2.24.0/glib/gslice.c x/glib-2.24.0/glib/gslice.c
--- x2/glib-2.24.0/glib/gslice.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gslice.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,7 +18,7 @@
  */
 /* MT safe */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gslice.h"
 
@@ -1126,7 +1126,7 @@
 #include <malloc.h>             /* memalign() */
 #endif
 
-/* from config.h:
+/* from config.h.win32:
  * define HAVE_POSIX_MEMALIGN           1 // if free(posix_memalign(3)) works, <stdlib.h>
  * define HAVE_COMPLIANT_POSIX_MEMALIGN 1 // if free(posix_memalign(3)) works for sizes != 2^n, <stdlib.h>
  * define HAVE_MEMALIGN                 1 // if free(memalign(3)) works, <malloc.h>
diff -Naur x2/glib-2.24.0/glib/gslist.c x/glib-2.24.0/glib/gslist.c
--- x2/glib-2.24.0/glib/gslist.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gslist.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gspawn-win32-helper.c x/glib-2.24.0/glib/gspawn-win32-helper.c
--- x2/glib-2.24.0/glib/gspawn-win32-helper.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gspawn-win32-helper.c	2013-04-12 12:32:22.000000000 +0200
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <fcntl.h>
 
diff -Naur x2/glib-2.24.0/glib/gspawn-win32.c x/glib-2.24.0/glib/gspawn-win32.c
--- x2/glib-2.24.0/glib/gspawn-win32.c	2009-08-28 05:51:21.000000000 +0200
+++ x/glib-2.24.0/glib/gspawn-win32.c	2013-04-12 12:32:22.000000000 +0200
@@ -42,7 +42,7 @@
 /* Define this to get some logging all the time */
 /* #define G_SPAWN_WIN32_DEBUG */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "gprintfint.h"
diff -Naur x2/glib-2.24.0/glib/gspawn.c x/glib-2.24.0/glib/gspawn.c
--- x2/glib-2.24.0/glib/gspawn.c	2009-08-28 05:51:21.000000000 +0200
+++ x/glib-2.24.0/glib/gspawn.c	2013-04-12 12:32:22.000000000 +0200
@@ -20,7 +20,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <sys/time.h>
 #include <sys/types.h>
diff -Naur x2/glib-2.24.0/glib/gstdio.c x/glib-2.24.0/glib/gstdio.c
--- x2/glib-2.24.0/glib/gstdio.c	2010-03-13 20:05:53.000000000 +0100
+++ x/glib-2.24.0/glib/gstdio.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,7 +18,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #define G_STDIO_NO_WRAP_ON_UNIX
 
diff -Naur x2/glib-2.24.0/glib/gstrfuncs.c x/glib-2.24.0/glib/gstrfuncs.c
--- x2/glib-2.24.0/glib/gstrfuncs.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gstrfuncs.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #define _GNU_SOURCE		/* For stpcpy */
 
diff -Naur x2/glib-2.24.0/glib/gstring.c x/glib-2.24.0/glib/gstring.c
--- x2/glib-2.24.0/glib/gstring.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gstring.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
diff -Naur x2/glib-2.24.0/glib/gtestutils.c x/glib-2.24.0/glib/gtestutils.c
--- x2/glib-2.24.0/glib/gtestutils.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gtestutils.c	2013-04-12 12:32:22.000000000 +0200
@@ -17,7 +17,7 @@
  * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  */
-#include "config.h"
+#include "config.h.win32"
 #include "gtestutils.h"
 #include <glib.h>
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gthread.c x/glib-2.24.0/glib/gthread.c
--- x2/glib-2.24.0/glib/gthread.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gthread.c	2013-04-12 12:32:22.000000000 +0200
@@ -38,7 +38,7 @@
 #define G_IMPLEMENT_INLINES 1
 #define __G_THREAD_C__
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "gthreadprivate.h"
@@ -1238,7 +1238,7 @@
  * a GMutex.  In that case, the first access we make to the static mutex
  * must first allocate the normal GMutex and store it into the pointer.
  *
- * configure.in writes macros into glibconfig.h to determine if
+ * configure.in writes macros into glibconfig.h.win32 to determine if
  * g_static_mutex_get_mutex() accesses the sturcture in memory directly
  * (on platforms where we are able to do that) or if it ends up here,
  * where we may have to allocate the GMutex before returning it.
@@ -1323,7 +1323,7 @@
   g_return_if_fail (mutex);
 
   /* The runtime_mutex is the first (or only) member of GStaticMutex,
-   * see both versions (of glibconfig.h) in configure.in. Note, that
+   * see both versions (of glibconfig.h.win32) in configure.in. Note, that
    * this variable is NULL, if g_thread_init() hasn't been called or
    * if we're using the default thread implementation and it provides
    * static mutexes. */
diff -Naur x2/glib-2.24.0/glib/gthreadpool.c x/glib-2.24.0/glib/gthreadpool.c
--- x2/glib-2.24.0/glib/gthreadpool.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gthreadpool.c	2013-04-12 12:32:23.000000000 +0200
@@ -24,7 +24,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gtimer.c x/glib-2.24.0/glib/gtimer.c
--- x2/glib-2.24.0/glib/gtimer.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gtimer.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,8 +28,8 @@
  * MT safe
  */
 
-#include "config.h"
-#include "glibconfig.h"
+#include "config.h.win32"
+#include "glibconfig.h.win32"
 
 #include <stdlib.h>
 
diff -Naur x2/glib-2.24.0/glib/gtree.c x/glib-2.24.0/glib/gtree.c
--- x2/glib-2.24.0/glib/gtree.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gtree.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "glib.h"
 #include "galias.h"
diff -Naur x2/glib-2.24.0/glib/gtypes.h x/glib-2.24.0/glib/gtypes.h
--- x2/glib-2.24.0/glib/gtypes.h	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gtypes.h	2013-04-12 12:33:22.000000000 +0200
@@ -31,7 +31,7 @@
 #ifndef __G_TYPES_H__
 #define __G_TYPES_H__
 
-#include <glibconfig.h>
+#include <glibconfig.h.win32>
 #include <glib/gmacros.h>
 
 G_BEGIN_DECLS
@@ -129,7 +129,7 @@
 
 /* Portable endian checks and conversions
  *
- * glibconfig.h defines G_BYTE_ORDER which expands to one of
+ * glibconfig.h.win32 defines G_BYTE_ORDER which expands to one of
  * the below macros.
  */
 #define G_LITTLE_ENDIAN 1234
@@ -308,7 +308,7 @@
     (((guint32) (val) & (guint32) 0x00ff00ffU) << 8) | \
     (((guint32) (val) & (guint32) 0xff00ff00U) >> 8)))
 
-/* The G*_TO_?E() macros are defined in glibconfig.h.
+/* The G*_TO_?E() macros are defined in glibconfig.h.win32.
  * The transformation is symmetric, so the FROM just maps to the TO.
  */
 #define GINT16_FROM_LE(val)	(GINT16_TO_LE (val))
diff -Naur x2/glib-2.24.0/glib/gunibreak.c x/glib-2.24.0/glib/gunibreak.c
--- x2/glib-2.24.0/glib/gunibreak.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gunibreak.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdlib.h>
 
diff -Naur x2/glib-2.24.0/glib/gunicollate.c x/glib-2.24.0/glib/gunicollate.c
--- x2/glib-2.24.0/glib/gunicollate.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gunicollate.c	2013-04-12 12:32:22.000000000 +0200
@@ -18,7 +18,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <locale.h>
 #include <string.h>
diff -Naur x2/glib-2.24.0/glib/gunidecomp.c x/glib-2.24.0/glib/gunidecomp.c
--- x2/glib-2.24.0/glib/gunidecomp.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/gunidecomp.c	2013-04-12 12:32:23.000000000 +0200
@@ -19,7 +19,7 @@
  *   Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdlib.h>
 
diff -Naur x2/glib-2.24.0/glib/guniprop.c x/glib-2.24.0/glib/guniprop.c
--- x2/glib-2.24.0/glib/guniprop.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/guniprop.c	2013-04-12 12:32:22.000000000 +0200
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdlib.h>
 #include <stddef.h>
diff -Naur x2/glib-2.24.0/glib/gurifuncs.c x/glib-2.24.0/glib/gurifuncs.c
--- x2/glib-2.24.0/glib/gurifuncs.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gurifuncs.c	2013-04-12 12:32:22.000000000 +0200
@@ -20,7 +20,7 @@
  * Author: Alexander Larsson <alexl@redhat.com>
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gurifuncs.h"
 
@@ -31,7 +31,7 @@
 
 #include <string.h>
 
-#include "config.h"
+#include "config.h.win32"
 #include "galias.h"
 
 /**
diff -Naur x2/glib-2.24.0/glib/gutf8.c x/glib-2.24.0/glib/gutf8.c
--- x2/glib-2.24.0/glib/gutf8.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gutf8.c	2013-04-12 12:32:22.000000000 +0200
@@ -19,7 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <stdlib.h>
 #ifdef HAVE_CODESET
diff -Naur x2/glib-2.24.0/glib/gutils.c x/glib-2.24.0/glib/gutils.c
--- x2/glib-2.24.0/glib/gutils.c	2010-02-11 07:34:48.000000000 +0100
+++ x/glib-2.24.0/glib/gutils.c	2013-04-12 12:32:22.000000000 +0200
@@ -28,7 +28,7 @@
  * MT safe for the unix part, FIXME: make the win32 part MT safe as well.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
diff -Naur x2/glib-2.24.0/glib/gutils.h x/glib-2.24.0/glib/gutils.h
--- x2/glib-2.24.0/glib/gutils.h	2010-02-11 07:34:48.000000000 +0100
+++ x/glib-2.24.0/glib/gutils.h	2013-04-12 12:33:22.000000000 +0200
@@ -61,7 +61,7 @@
 #endif /* !G_OS_WIN32 */
 
 /* Define G_VA_COPY() to do the right thing for copying va_list variables.
- * glibconfig.h may have already defined G_VA_COPY as va_copy or __va_copy.
+ * glibconfig.h.win32 may have already defined G_VA_COPY as va_copy or __va_copy.
  */
 #if !defined (G_VA_COPY)
 #  if defined (__GNUC__) && defined (__PPC__) && (defined (_CALL_SYSV) || defined (_WIN32))
@@ -76,7 +76,7 @@
 /* inlining hassle. for compilers that don't allow the `inline' keyword,
  * mostly because of strict ANSI C compliance or dumbness, we try to fall
  * back to either `__inline__' or `__inline'.
- * G_CAN_INLINE is defined in glibconfig.h if the compiler seems to be 
+ * G_CAN_INLINE is defined in glibconfig.h.win32 if the compiler seems to be 
  * actually *capable* to do function inlining, in which case inline 
  * function bodies do make sense. we also define G_INLINE_FUNC to properly 
  * export the function prototypes if no inlining can be performed.
diff -Naur x2/glib-2.24.0/glib/gvariant-serialiser.c x/glib-2.24.0/glib/gvariant-serialiser.c
--- x2/glib-2.24.0/glib/gvariant-serialiser.c	2010-03-26 13:34:20.000000000 +0100
+++ x/glib-2.24.0/glib/gvariant-serialiser.c	2013-04-12 12:32:22.000000000 +0200
@@ -21,7 +21,7 @@
  */
 
 /* Prologue {{{1 */
-#include "config.h"
+#include "config.h.win32"
 
 #include "gvariant-serialiser.h"
 
diff -Naur x2/glib-2.24.0/glib/gvariant.c x/glib-2.24.0/glib/gvariant.c
--- x2/glib-2.24.0/glib/gvariant.c	2010-03-26 13:34:20.000000000 +0100
+++ x/glib-2.24.0/glib/gvariant.c	2013-04-12 12:32:22.000000000 +0200
@@ -22,7 +22,7 @@
 
 /* Prologue {{{1 */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include <glib/gvariant-serialiser.h>
 #include "gvariant-internal.h"
diff -Naur x2/glib-2.24.0/glib/gvarianttype.c x/glib-2.24.0/glib/gvarianttype.c
--- x2/glib-2.24.0/glib/gvarianttype.c	2010-03-17 21:55:34.000000000 +0100
+++ x/glib-2.24.0/glib/gvarianttype.c	2013-04-12 12:32:22.000000000 +0200
@@ -20,7 +20,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gvarianttype.h"
 
diff -Naur x2/glib-2.24.0/glib/gvarianttypeinfo.c x/glib-2.24.0/glib/gvarianttypeinfo.c
--- x2/glib-2.24.0/glib/gvarianttypeinfo.c	2010-03-26 13:34:20.000000000 +0100
+++ x/glib-2.24.0/glib/gvarianttypeinfo.c	2013-04-12 12:32:22.000000000 +0200
@@ -20,7 +20,7 @@
  * Author: Ryan Lortie <desrt@desrt.ca>
  */
 
-#include "config.h"
+#include "config.h.win32"
 
 #include "gvarianttypeinfo.h"
 
diff -Naur x2/glib-2.24.0/glib/gwin32.c x/glib-2.24.0/glib/gwin32.c
--- x2/glib-2.24.0/glib/gwin32.c	2010-02-11 07:34:48.000000000 +0100
+++ x/glib-2.24.0/glib/gwin32.c	2013-04-12 12:32:22.000000000 +0200
@@ -29,9 +29,9 @@
  * MT safe for the unix part, FIXME: make the win32 part MT safe as well.
  */
 
-#include "config.h"
+#include "config.h.win32"
 
-#include "glibconfig.h"
+#include "glibconfig.h.win32"
 
 #include <stdlib.h>
 #include <stdio.h>
diff -Naur x2/glib-2.24.0/glib/libcharset/localcharset.c x/glib-2.24.0/glib/libcharset/localcharset.c
--- x2/glib-2.24.0/glib/libcharset/localcharset.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/libcharset/localcharset.c	2013-04-12 12:32:23.000000000 +0200
@@ -19,7 +19,7 @@
 
 /* Written by Bruno Haible <bruno@clisp.org>.  */
 
-#include "config.h"
+#include "config.h.win32"
 
 /* Specification.  */
 #include "localcharset.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_chartables.c x/glib-2.24.0/glib/pcre/pcre_chartables.c
--- x2/glib-2.24.0/glib/pcre/pcre_chartables.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_chartables.c	2013-04-12 12:32:22.000000000 +0200
@@ -21,7 +21,7 @@
 unit might reference this" and so it will always be supplied to the linker. */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_compile.c x/glib-2.24.0/glib/pcre/pcre_compile.c
--- x2/glib-2.24.0/glib/pcre/pcre_compile.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_compile.c	2013-04-12 12:32:22.000000000 +0200
@@ -43,7 +43,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #define NLBLOCK cd             /* Block containing newline information */
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_config.c x/glib-2.24.0/glib/pcre/pcre_config.c
--- x2/glib-2.24.0/glib/pcre/pcre_config.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_config.c	2013-04-12 12:32:22.000000000 +0200
@@ -42,7 +42,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_dfa_exec.c x/glib-2.24.0/glib/pcre/pcre_dfa_exec.c
--- x2/glib-2.24.0/glib/pcre/pcre_dfa_exec.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_dfa_exec.c	2013-04-12 12:32:22.000000000 +0200
@@ -45,7 +45,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #define NLBLOCK md             /* Block containing newline information */
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_exec.c x/glib-2.24.0/glib/pcre/pcre_exec.c
--- x2/glib-2.24.0/glib/pcre/pcre_exec.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_exec.c	2013-04-12 12:32:22.000000000 +0200
@@ -43,7 +43,7 @@
 possible. There are also some static supporting functions. */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #define NLBLOCK md             /* Block containing newline information */
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_fullinfo.c x/glib-2.24.0/glib/pcre/pcre_fullinfo.c
--- x2/glib-2.24.0/glib/pcre/pcre_fullinfo.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_fullinfo.c	2013-04-12 12:32:22.000000000 +0200
@@ -43,7 +43,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_get.c x/glib-2.24.0/glib/pcre/pcre_get.c
--- x2/glib-2.24.0/glib/pcre/pcre_get.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_get.c	2013-04-12 12:32:22.000000000 +0200
@@ -44,7 +44,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_globals.c x/glib-2.24.0/glib/pcre/pcre_globals.c
--- x2/glib-2.24.0/glib/pcre/pcre_globals.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_globals.c	2013-04-12 12:32:22.000000000 +0200
@@ -47,7 +47,7 @@
 differently, and global variables are not used (see pcre.in). */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_info.c x/glib-2.24.0/glib/pcre/pcre_info.c
--- x2/glib-2.24.0/glib/pcre/pcre_info.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_info.c	2013-04-12 12:32:22.000000000 +0200
@@ -44,7 +44,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_internal.h x/glib-2.24.0/glib/pcre/pcre_internal.h
--- x2/glib-2.24.0/glib/pcre/pcre_internal.h	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_internal.h	2013-04-12 12:33:22.000000000 +0200
@@ -307,7 +307,7 @@
 defined here.
 
 The macros are controlled by the value of LINK_SIZE. This defaults to 2 in
-the config.h file, but can be overridden by using -D on the command line. This
+the config.h.win32 file, but can be overridden by using -D on the command line. This
 is automated on Unix systems via the "configure" command. */
 
 #if LINK_SIZE == 2
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_maketables.c x/glib-2.24.0/glib/pcre/pcre_maketables.c
--- x2/glib-2.24.0/glib/pcre/pcre_maketables.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_maketables.c	2013-04-12 12:32:22.000000000 +0200
@@ -46,7 +46,7 @@
 
 #ifndef DFTABLES
 #  ifdef HAVE_CONFIG_H
-#  include "config.h"
+#  include "config.h.win32"
 #  endif
 #  include "pcre_internal.h"
 #endif
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_newline.c x/glib-2.24.0/glib/pcre/pcre_newline.c
--- x2/glib-2.24.0/glib/pcre/pcre_newline.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_newline.c	2013-04-12 12:32:22.000000000 +0200
@@ -48,7 +48,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_ord2utf8.c x/glib-2.24.0/glib/pcre/pcre_ord2utf8.c
--- x2/glib-2.24.0/glib/pcre/pcre_ord2utf8.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_ord2utf8.c	2013-04-12 12:32:22.000000000 +0200
@@ -42,7 +42,7 @@
 character value into a UTF8 string. */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_refcount.c x/glib-2.24.0/glib/pcre/pcre_refcount.c
--- x2/glib-2.24.0/glib/pcre/pcre_refcount.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_refcount.c	2013-04-12 12:32:22.000000000 +0200
@@ -45,7 +45,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_study.c x/glib-2.24.0/glib/pcre/pcre_study.c
--- x2/glib-2.24.0/glib/pcre/pcre_study.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_study.c	2013-04-12 12:32:22.000000000 +0200
@@ -43,7 +43,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_tables.c x/glib-2.24.0/glib/pcre/pcre_tables.c
--- x2/glib-2.24.0/glib/pcre/pcre_tables.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_tables.c	2013-04-12 12:32:22.000000000 +0200
@@ -45,7 +45,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_try_flipped.c x/glib-2.24.0/glib/pcre/pcre_try_flipped.c
--- x2/glib-2.24.0/glib/pcre/pcre_try_flipped.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_try_flipped.c	2013-04-12 12:32:22.000000000 +0200
@@ -44,7 +44,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_ucp_searchfuncs.c x/glib-2.24.0/glib/pcre/pcre_ucp_searchfuncs.c
--- x2/glib-2.24.0/glib/pcre/pcre_ucp_searchfuncs.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_ucp_searchfuncs.c	2013-04-12 12:32:22.000000000 +0200
@@ -44,7 +44,7 @@
 properties. */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_valid_utf8.c x/glib-2.24.0/glib/pcre/pcre_valid_utf8.c
--- x2/glib-2.24.0/glib/pcre/pcre_valid_utf8.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_valid_utf8.c	2013-04-12 12:32:22.000000000 +0200
@@ -1,4 +1,4 @@
-#include "config.h"
+#include "config.h.win32"
 #include "pcre_internal.h"
 
 /*
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_version.c x/glib-2.24.0/glib/pcre/pcre_version.c
--- x2/glib-2.24.0/glib/pcre/pcre_version.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_version.c	2013-04-12 12:32:22.000000000 +0200
@@ -43,7 +43,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/pcre/pcre_xclass.c x/glib-2.24.0/glib/pcre/pcre_xclass.c
--- x2/glib-2.24.0/glib/pcre/pcre_xclass.c	2009-04-01 01:04:20.000000000 +0200
+++ x/glib-2.24.0/glib/pcre/pcre_xclass.c	2013-04-12 12:32:22.000000000 +0200
@@ -44,7 +44,7 @@
 
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/update-pcre/pcre_ucp_searchfuncs.c x/glib-2.24.0/glib/update-pcre/pcre_ucp_searchfuncs.c
--- x2/glib-2.24.0/glib/update-pcre/pcre_ucp_searchfuncs.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/update-pcre/pcre_ucp_searchfuncs.c	2013-04-12 12:32:22.000000000 +0200
@@ -44,7 +44,7 @@
 properties. */
 
 #ifdef HAVE_CONFIG_H
-#include "config.h"
+#include "config.h.win32"
 #endif
 
 #include "pcre_internal.h"
diff -Naur x2/glib-2.24.0/glib/update-pcre/pcre_valid_utf8.c x/glib-2.24.0/glib/update-pcre/pcre_valid_utf8.c
--- x2/glib-2.24.0/glib/update-pcre/pcre_valid_utf8.c	2009-08-29 04:52:23.000000000 +0200
+++ x/glib-2.24.0/glib/update-pcre/pcre_valid_utf8.c	2013-04-12 12:32:22.000000000 +0200
@@ -1,4 +1,4 @@
-#include "config.h"
+#include "config.h.win32"
 #include "pcre_internal.h"
 
 /*
diff -Naur x2/unistd.c x/unistd.c
--- x2/unistd.c	1970-01-01 01:00:00.000000000 +0100
+++ x/unistd.c	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,506 @@
+/* This file is part of the KDE project
+   Copyright (C) 2003-2004 Jaroslaw Staniek <js@iidea.pl>
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public License
+   along with this program; see the file COPYING.  If not, write to
+   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110-1301, USA.
+*/
+
+#define _WIN32_WINNT 0x0500
+#include <winsock2.h>
+#include <windows.h>
+
+#include <direct.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// BEGIN stat.c
+int lstat(const char *path, struct stat *sb)
+{
+  return _stat(path,(struct _stat*)sb);
+}
+
+int fchmod(int __fd, mode_t __mode)
+{
+  return 0;
+}
+// END stat.c
+
+int getgroups(int size, gid_t list[])
+{
+	/* TODO */
+	return 0;
+}
+
+int readlink(const char *__path, char *__buf, int __buflen)
+{
+    if (!__path) {
+      errno = EINVAL;
+      return -1;
+    }
+    if ( (__buflen < 0) || ((int)strlen(__path)>(__buflen-1)) )
+    {
+      errno = ENAMETOOLONG;
+      return -1;
+    }
+    if (access(__path, R_OK) == 0) {
+      /* ok, copy to buf */
+      strncpy(__buf,__path,__buflen);
+      errno = 0;
+      return 0;
+    }
+    errno = ENOENT;
+    return -1;
+}
+
+int symlink(const char *__name1, const char *__name2)
+{
+	if(!CopyFileA(__name1, __name2, FALSE)) {
+	    switch(GetLastError()) {
+	        case ERROR_FILE_EXISTS:
+    	        errno = EEXIST;
+    	        break;
+    	    case ERROR_CANNOT_MAKE:
+    	        errno = EACCES;
+    	        break;
+    	    default:
+    	        errno = EFAULT;
+    	        break;
+        }
+	    return -1; 
+	}
+	return 0;
+}
+
+int link(const char *__name1, const char *__name2)
+{
+	return symlink(__name1, __name2);
+}
+
+int chown(const char *__path, uid_t __owner, gid_t __group)
+{ 
+  return 0; 
+}
+
+int fchown(int __fd, uid_t __owner, gid_t __group )
+{
+  return 0; 
+}
+
+/* Get the real user ID of the calling process.  */
+uid_t getuid()
+{
+  return -2;    // keep in sync with Qt!
+}
+
+/* Get the effective user ID of the calling process.  */
+uid_t geteuid (void)
+{
+  return -2;    // keep in sync with Qt!
+}
+
+/* Get the real group ID of the calling process.  */
+gid_t getgid (void)
+{
+  return -2;    // keep in sync with Qt!
+}
+
+/* Get the effective group ID of the calling process.  */
+gid_t getegid (void)
+{
+  return -2;    // keep in sync with Qt!
+}
+
+int pipe(int *fd)
+{
+  /** @todo */
+  return _pipe( fd, 256, O_BINARY ); /* OK? */
+}
+
+pid_t fork(void)
+{
+  /** @todo */
+  return -1;
+}
+
+pid_t setsid(void)
+{
+  /** @todo */
+  return -1;
+}
+
+unsigned alarm(unsigned __secs )
+{
+  /** @todo */
+  return 1;
+}
+
+//typedef __int64 size_t;
+
+/*#define INCL_WINSOCK_API_PROTOTYPES 0
+#include <winsock2.h>*/
+
+// using winsock gethostname(), which would be taken normally requires WSAStartup called before 
+// which will not be done in every case
+//http://msdn.microsoft.com/library/default.asp?url=/library/en-us/winsock/winsock/gethostname_2.asp
+
+int kde_gethostname(char *__name, size_t __len)
+{
+  DWORD len = __len;
+  if (0==GetComputerNameA(__name, &len))
+    return -1;
+  return 0;
+}
+
+#define getlogin_buf_size 255
+char getlogin_buf[getlogin_buf_size+1];
+
+char* getlogin()
+{
+/*! @todo make this reentrant!*/
+	size_t size = sizeof(getlogin_buf);
+	*getlogin_buf = 0;
+	if (!GetUserNameA(getlogin_buf, (LPDWORD)&size))
+		return 0;
+	return getlogin_buf;
+}
+
+int usleep(unsigned int usec)
+{
+	Sleep(usec/1000);
+}
+
+int sleep(unsigned int sec)
+{
+	Sleep(sec*1000);
+}
+
+
+int setreuid(uid_t ruid, uid_t euid)
+{
+	/*! @todo */
+	return 0;
+}
+
+
+/* this is based on code taken from the GNU libc, distributed under the LGPL license */
+
+/* Generate a unique temporary file name from TEMPLATE.
+
+   TEMPLATE has the form:
+
+   <path>/ccXXXXXX<suffix>
+
+   SUFFIX_LEN tells us how long <suffix> is (it can be zero length).
+
+   The last six characters of TEMPLATE before <suffix> must be "XXXXXX";
+   they are replaced with a string that makes the filename unique.
+
+   Returns a file descriptor open on the file for reading and writing.  */
+
+int mkstemps (char* _template, int suffix_len)
+{
+  static const char letters[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  char *XXXXXX;
+  int len;
+  int count;
+  int value;
+
+  len = strlen (_template);
+
+  if ((int) len < 6 + suffix_len || strncmp (&_template[len - 6 - suffix_len], "XXXXXX", 6))
+      return -1;
+
+  XXXXXX = &_template[len - 6 - suffix_len];
+
+  value = rand();
+  for (count = 0; count < 256; ++count)
+  {
+      int v = value;
+      int fd;
+
+      /* Fill in the random bits.  */
+      XXXXXX[0] = letters[v % 62];
+      v /= 62;
+      XXXXXX[1] = letters[v % 62];
+      v /= 62;
+      XXXXXX[2] = letters[v % 62];
+      v /= 62;
+      XXXXXX[3] = letters[v % 62];
+      v /= 62;
+      XXXXXX[4] = letters[v % 62];
+      v /= 62;
+      XXXXXX[5] = letters[v % 62];
+
+      fd = open (_template, O_RDWR|O_CREAT|O_EXCL, 0600);
+      if (fd >= 0)
+		/* The file does not exist.  */
+		return fd;
+
+      /* This is a random value.  It is only necessary that the next
+	 TMP_MAX values generated by adding 7777 to VALUE are different
+	 with (module 2^32).  */
+      value += 7777;
+    }
+  /* We return the null string if we can't find a unique file name.  */
+  _template[0] = '\0';
+  return -1;
+}
+
+// from kdecore/fakes.c
+int mkstemp (char* _template)
+{
+  return mkstemps( _template, 0 );
+}
+
+// from kdecore/fakes.c
+int seteuid(uid_t euid)
+{
+    return setreuid(-1, euid); /* Well, if you have neither you are in trouble :) */
+}
+
+// from kdecore/fakes.c
+/* Generate a unique temporary directory name from TEMPLATE.
+
+   TEMPLATE has the form:
+
+   <path>/ccXXXXXX
+
+
+   The last six characters of TEMPLATE must be "XXXXXX";
+   they are replaced with a string that makes the filename unique.
+
+   Returns a file descriptor open on the file for reading and writing.  */
+
+char* mkdtemp (char* _template)
+{
+  static const char letters[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  char *XXXXXX;
+  int len;
+  int count;
+  int value;
+
+  len = strlen (_template);
+
+  if ((int) len < 6 || strncmp (&_template[len - 6], "XXXXXX", 6))
+      return 0;
+
+  XXXXXX = &_template[len - 6];
+
+  value = rand();
+  for (count = 0; count < 256; ++count)
+  {
+      int v = value;
+
+      /* Fill in the random bits.  */
+      XXXXXX[0] = letters[v % 62];
+      v /= 62;
+      XXXXXX[1] = letters[v % 62];
+      v /= 62;
+      XXXXXX[2] = letters[v % 62];
+      v /= 62;
+      XXXXXX[3] = letters[v % 62];
+      v /= 62;
+      XXXXXX[4] = letters[v % 62];
+      v /= 62;
+      XXXXXX[5] = letters[v % 62];
+
+      /* This is a random value.  It is only necessary that the next
+	 TMP_MAX values generated by adding 7777 to VALUE are different
+	 with (module 2^32).  */
+      value += 7777;
+
+      if (!mkdir(_template))
+		return _template;	
+    }
+    return 0;
+}
+
+int revoke(const char *tty)
+{
+        errno = ENOTSUP;
+        return -1;
+}
+
+/* getpagesize for windows */
+//static 
+long getpagesize (void)
+{
+    static long g_pagesize = 0;
+    if (! g_pagesize) {
+        SYSTEM_INFO system_info;
+        GetSystemInfo (&system_info);
+        g_pagesize = system_info.dwPageSize;
+    }
+    return g_pagesize;
+}
+
+#ifndef __MINGW32__
+// XGetopt.cpp  Version 1.2
+//
+// Author:  Hans Dietrich
+//          hdietrich2@hotmail.com
+//
+// Description:
+//     XGetopt.cpp implements getopt(), a function to parse command lines.
+//
+// History
+//     Version 1.2 - 2003 May 17
+//     - Added Unicode support
+//
+//     Version 1.1 - 2002 March 10
+//     - Added example to XGetopt.cpp module header 
+//
+// This software is released into the public domain.
+// You are free to use it in any way you like.
+//
+// This software is provided "as is" with no expressed
+// or implied warranty.  I accept no liability for any
+// damage or loss of business that this software may cause.
+//
+// CE: from http://www.codeproject.com/cpp/xgetopt.asp
+//     removed unicode support to compile with mingw
+char *optarg = NULL;	// global argument pointer
+int   optind = 0; 	// global argv index
+int   opterr = 0; 	
+
+int getopt(int argc, char **argv, const char *optstring)
+{
+	static char *next = NULL;
+	char c = '?';
+	char *cp = NULL;
+	if (optind == 0)
+		next = NULL;
+
+	optarg = NULL;
+
+	if (next == NULL || *next == '\0')
+	{
+		if (optind == 0)
+			optind++;
+
+		if (optind >= argc || argv[optind][0] != '-' || argv[optind][1] == '\0')
+		{
+			optarg = NULL;
+			if (optind < argc)
+				optarg = argv[optind];
+			return -1;
+		}
+
+		if (strcmp(argv[optind], "--") == 0)
+		{
+			optind++;
+			optarg = NULL;
+			if (optind < argc)
+				optarg = argv[optind];
+			return -1;
+		}
+
+		next = argv[optind];
+		next++;		// skip past -
+		optind++;
+	}
+
+	c = *next++;
+	cp = strchr(optstring, c);
+
+	if (cp == NULL || c == ':')
+		return '?';
+
+	cp++;
+	if (*cp == ':')
+	{
+		if (*next != '\0')
+		{
+			optarg = next;
+			next = NULL;
+		}
+		else if (optind < argc)
+		{
+			optarg = argv[optind];
+			optind++;
+		}
+		else
+		{
+			return '?';
+		}
+	}
+
+	return c;
+}
+#endif  // __MINGW32__
+
+
+int truncate(const char *path, off_t length)
+{
+    HANDLE hFile;
+    LARGE_INTEGER fileSize;
+
+    if( length < 0 )
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    hFile = CreateFileA( path, GENERIC_WRITE, 0, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0 );
+
+    if( hFile == INVALID_HANDLE_VALUE )
+    {
+        DWORD dwLastErr = GetLastError();
+        errno = EACCES;
+        return -1;
+    }
+
+    if( GetFileSizeEx( hFile, &fileSize ) == 0 )
+    {
+        DWORD dwLastErr = GetLastError();
+        CloseHandle( hFile );
+        errno = EACCES;
+        return -1;
+    }
+
+    if( fileSize.QuadPart < length)
+    {
+        CloseHandle( hFile );
+        errno = EINVAL;
+        return -1;
+    }
+
+    fileSize.QuadPart = length;
+
+    if( SetFilePointerEx( hFile, fileSize, 0, FILE_BEGIN ) == 0 )
+    {
+        DWORD dwLastErr = GetLastError();
+        CloseHandle( hFile );
+        errno = EACCES;
+        return -1;
+    }
+
+    CloseHandle( hFile );
+    return 0;
+}
+
+#ifndef __MINGW32__
+// TODO (js) consider using _chsize_s
+int ftruncate(int fd, off_t length)
+{
+  return _chsize (fd, length);
+}
+#endif  // __MINGW32__
+
diff -Naur x2/unistd.h x/unistd.h
--- x2/unistd.h	1970-01-01 01:00:00.000000000 +0100
+++ x/unistd.h	2013-04-12 12:24:43.000000000 +0200
@@ -0,0 +1,62 @@
+/* This file is part of the KDE project
+   Copyright (C) 2003-2004 Jaroslaw Staniek <staniek@kde.org>
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public License
+   along with this program; see the file COPYING.  If not, write to
+   the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110-1301, USA.
+*/
+
+#ifndef SMI_UNISTD_H
+#define SMI_UNISTD_H
+
+#define environ _environ
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern int optind;
+
+#define	F_OK	0
+#define	R_OK	4
+#define	W_OK	2
+#define	X_OK	1 
+
+#ifndef STDIN_FILENO
+#define STDIN_FILENO 0
+#endif
+
+#ifndef STDOUT_FILENO
+#define STDOUT_FILENO 1
+#endif
+
+#ifndef STDERR_FILENO
+#define STDERR_FILENO 2
+#endif
+
+#if _MSC_VER < 1600
+#define ENOTSUP       ENOSYS
+#endif
+// from sys/types.h:
+#ifndef __MINGW32__
+typedef int mode_t;
+#endif
+typedef unsigned int gid_t;
+typedef unsigned int uid_t;
+typedef int pid_t;
+#ifdef __cplusplus
+}
+#endif
+
+#endif // KDEWIN_UNISTD_H
